packages/@okta/vuepress-site/docs/guides/sign-into-spa-redirect/main/react/configmid.md

```diff
@@ -16,100 +15,70 @@ The Okta React SDK requires an instance of an `OktaAuth` object with configurati
      scopes: ['openid', 'profile', 'email', 'offline_access']
    });
 
-   class App extends Component {
-
-     constructor(props) {
-       super(props);
-       this.restoreOriginalUri = async (_oktaAuth, originalUri) => {
-         props.history.replace(toRelativeUrl(originalUri || '/', window.location.origin));
-       };
-     }
-
-     render() {
-       return (
-         <Security oktaAuth={oktaAuth} restoreOriginalUri={this.restoreOriginalUri}>
-           <Route path="/" exact={true} component={Home}/>
-           <Route path="/login/callback" component={LoginCallback}/>
+   function App() {
+     const history = useHistory();
+     const restoreOriginalUri = (_oktaAuth,  originalUri) => {
+       history.replace(toRelativeUrl(originalUri || '/', window.location.origin));
+     };
+
+     return (
+       <>
+         <Security oktaAuth={oktaAuth} restoreOriginalUri={restoreOriginalUri}>
+           <Switch>
+               <Route path="/" exact component={Home}/>
+               <Route path="/login/callback" component={LoginCallback}/>
+           </Switch>
          </Security>
-       );
-     }
-   }
-
-   const AppWithRouterAccess = withRouter(App);
-
-   class RouterApp extends Component {
-     render() {
-       return (<Router><AppWithRouterAccess/></Router>);
-     }
+       </>
+     );
    }
 
-   export default RouterApp;
+   export default App
    ```
 
-2. Install `react-router-dom` into your app with the following command:
+1. Ensure that your default custom authorization server has an access policy. Add an access policy if it's not there. See [Create access polices](https://help.okta.com/okta_help.htm?type=oie&id=ext-create-access-policies).
+
+1. Install `react-router-dom` into your app with the following command:
 
    ```shell
-   npm install react-router-dom@5
+   npm install react-router-dom@5.3.4
    ```
 
-   > **Note**: This example only works with React Router v5. There's an [open issue](https://github.com/okta/okta-react/issues/178) to support React Router v6.
+1. Update `src/main.jsx` to add routing:
 
-3. Replace the `clientId` and `issuer` placeholder values with the values that you obtained earlier.
+   ```jsx
+   import ReactDOM from 'react-dom/client'
+   import { BrowserRouter } from 'react-router-dom'
+   import App from './App.jsx'
+   import './index.css'
+
+   ReactDOM.createRoot(document.getElementById('root')).render(
+     <BrowserRouter>
+       <App />
+     </BrowserRouter>,
+   )
+   ```
 
-4. Add a `src/Home.js` file that renders login and logout buttons:
+1. Add a `src/Home.jsx` file that renders the starting view for the application:
```

Comment:
```suggestion
1. Add a `src/Home.jsx` file that renders the starting view for the app:
```

Reply:
Great catch ... maybe I can use the "Sign In" and "Sign Out" buttons, which is what we actually implement in the code.

Reply:
Actually, going with your original edit, as the dev advocate made the original change. And the buttons come in later steps.